frontend/packages/pipelines-plugin/src/components/quicksearch/PipelineQuickSearch.tsx

```diff
@@ -27,17 +37,60 @@ const Contents: React.FC<{
 }) => {
   const { t } = useTranslation();
   const savedCallback = React.useRef(null);
+  const [failedTasks, setFailedTasks] = React.useState<string[]>([]);
+
+  useMetadataCleanup();
+  useMetadataFailureCleanup(failedTasks, (taskName) =>
+    setFailedTasks(failedTasks.filter((ft) => ft !== taskName)),
+  );
 
   React.useEffect(() => {
     savedCallback.current = callback;
   }, [callback]);
 
-  const catalogServiceItems = catalogService.items.map((service) => {
-    service.cta.callback = () => {
-      savedCallback.current(service.data);
+  const catalogServiceItems = catalogService.items.reduce((acc, item) => {
+    const installedTask = findInstalledTask(catalogService.items, item);
+
+    if (item.provider === TektonTaskProviders.community) {
+      item.attributes.installed = '';
+      if (installedTask) {
+        const installedVersion = item.attributes?.versions?.find(
+          (v) => v.version === installedTask.attributes?.versions[0]?.version,
+        );
+        item.attributes.installed = installedVersion.id.toString();
+      }
+    }
+
+    item.cta.callback = ({ selectedVersion }) => {
+      return new Promise((resolve) => {
+        if (item.provider === TektonTaskProviders.community) {
+          const selectedVersionUrl = getSelectedVersionUrl(item, selectedVersion);
+          if (installedTask) {
+            if (selectedVersion === item.attributes.installed) {
+              resolve(savedCallback.current(installedTask.data));
+            } else {
+              resolve(savedCallback.current({ metadata: { name: item.data.name } }));
+              updateTask(selectedVersionUrl, installedTask, namespace, item.data.name).catch(() =>
+                setFailedTasks([...failedTasks, item.data.name]),
+              );
+            }
+          } else {
+            resolve(savedCallback.current({ metadata: { name: item.data.name } }));
+            createTask(selectedVersionUrl, namespace).catch(() =>
```

Comment:
no action needed on `then` here only on `catch`?

Reply:
Yes, In case of successful task creation, the resources will be created and pulled in automatically by K8swatchers.